src/material/button/BUILD.bazel

```diff
@@ -16,51 +16,100 @@ ng_module(
     name = "button",
     srcs = glob(
         ["**/*.ts"],
-        exclude = ["**/*.spec.ts"],
+        exclude = [
+            "**/*.spec.ts",
+        ],
     ),
-    assets = [":button.css"] + glob(["**/*.html"]),
+    assets = [
+        ":button_high_contrast_scss",
+        ":button_scss",
+        ":fab_scss",
+        ":icon-button_scss",
+    ] + glob(["**/*.html"]),
     deps = [
-        "//src/cdk/a11y",
+        "//src/cdk/platform",
         "//src/material/core",
-        "@npm//@angular/animations",
-        "@npm//@angular/core",
-        "@npm//@angular/platform-browser",
     ],
 )
 
 sass_library(
     name = "button_scss_lib",
     srcs = glob(["**/_*.scss"]),
-    deps = ["//src/material/core:core_scss_lib"],
+    deps = [
+        "//:mdc_sass_lib",
+        "//src/material/core:core_scss_lib",
+    ],
+)
+
+sass_library(
+    name = "button_base_scss_lib",
+    srcs = ["_button-base.scss"],
 )
 
 sass_binary(
     name = "button_scss",
     src = "button.scss",
     deps = [
-        ":button_scss_lib",
+        ":button_base_scss_lib",
+        "//:mdc_sass_lib",
+        "//src/material:sass_lib",
+        "//src/material/core:core_scss_lib",
+    ],
+)
+
+sass_binary(
+    name = "button_high_contrast_scss",
+    src = "button-high-contrast.scss",
+    deps = [
         "//src/cdk:sass_lib",
+    ],
+)
+
+sass_binary(
+    name = "fab_scss",
+    src = "fab.scss",
+    deps = [
+        ":button_base_scss_lib",
+        "//:mdc_sass_lib",
+        "//src/material:sass_lib",
+        "//src/material/core:core_scss_lib",
+    ],
+)
+
+sass_binary(
+    name = "icon-button_scss",
+    src = "icon-button.scss",
+    deps = [
+        ":button_base_scss_lib",
+        "//:mdc_sass_lib",
+        "//src/material:sass_lib",
         "//src/material/core:core_scss_lib",
     ],
 )
 
 ng_test_library(
-    name = "unit_test_sources",
+    name = "button_tests_lib",
     srcs = glob(
         ["**/*.spec.ts"],
-        exclude = ["**/*.e2e.spec.ts"],
+        exclude = [
+            "**/*.e2e.spec.ts",
+        ],
     ),
     deps = [
         ":button",
+        "//src/cdk/platform",
         "//src/cdk/testing/private",
         "//src/material/core",
+        "//src/material/legacy-button",
```

Comment:
One step that got added to the process at some point is to invert this dependency by moving all of the base classes from legacy-button/ to button/, but I think this can come in a followup CL so we don't hold back moving other components

Reply:
Okay I will do this in a follow up PR